components/composition/Grid/Grid.ts

```diff
@@ -0,0 +1,98 @@
+import { LitElement, css, html, unsafeCSS } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+const directions = ['column', 'row-reverse', 'column-reverse'];
+
+const justify = [
+  'flex-end',
+  'center',
+  'space-between',
+  'space-around',
+  'space-evenly',
+];
+
+const align = ['flex-end', 'center'];
+
+const gap = ['xs', 'sm', 'lg', 'xl'];
+
+@customElement('diamond-grid')
+export class Grid extends LitElement {
+  @property({ type: Boolean }) readonly wrap?: boolean;
+  @property({ type: Boolean }) readonly inline?: boolean;
+  @property() readonly direction?:
+    | 'row'
+    | 'column'
+    | 'row-reverse'
+    | 'column-reverse' = 'row';
+  @property() readonly justifyContent?:
+    | 'flex-start'
+    | 'flex-end'
+    | 'center'
+    | 'space-between'
+    | 'space-around'
+    | 'space-evenly' = 'flex-start';
+  @property() readonly alignItems?: 'flex-start' | 'flex-end' | 'center' =
+    'flex-start';
+
+  static styles = css`
+    :host {
+      --diamond-grid-gap: var(--diamond-spacing);
+      display: flex;
+      margin: calc(var(--diamond-grid-gap) * -0.5);
+    }
+
+    :host([inline]) {
+      display: inline-flex;
+    }
+
+    :host([wrap]) {
+      flex-wrap: wrap;
+    }
+
+    ${unsafeCSS(
+      directions
+        .map(
+          (direction) =>
+            `:host([direction="${direction}"]) { flex-direction: ${direction}; }`,
+        )
+        .join('\n'),
+    )}
```

Comment:
Could we stop repeating the join using a wrapper fn

```
function cssMap(arr, callback) {
  return unsafeCSS(arr.map(callback).join('\n'));
}
```
Then this would be
```
${cssMap(directions, (direction) => `:host([direction="${direction}"]) { flex-direction: ${direction}; }`)};
```